core/arch/mangle_shared.c

```diff
@@ -430,7 +430,7 @@ insert_meta_call_vargs(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
              * which specifies that the caller must save xax (and xcx and xdx)
              */
             insert_get_mcontext_base(dcontext, ilist, instr, SCRATCH_REG0);
```

Comment:
I realized that this is wrong. `SCRATCH_REG0` is `r0` on ARM, which is the first argument to each clean call... x86 is clearly fine though. I'm curious, how does this work for Aarch64 either then? Isn't `x0` the first argument to clean calls as well?

@derekbruening thoughts?

Reply:
I could not find a test that enables profile_pcs on AArch64. We should definitely add some. This looks wrong, as the clobbering of `r0`/`x0` happens after the parameters are moved to the registers. I think we could use the link register here on ARM and AArch64, as we use `blx` to branch to the function. If we only allow AAPCS64 calls on AArch64, we should be able to use any of  the temporary registers (r9...r15), which the caller has to save & restore. 

Also, if we limit clean calls to AAPCS64 complaint calls on AArch64, we could also avoid saving/restoring r19..r28?

Reply:
@egrimley may have some thoughts as well

Reply:
Using the link reg seems to work on ARM32, I'll update the PR accordingly. Thanks!

Reply:
Sorry, I only evaluated whether ARM would be different from A64 and assumed the A64 code was correct without thinking about it much.